src/compiler/checker.ts

```diff
@@ -20018,7 +20018,7 @@ namespace ts {
                     // We clone the contextual mapper to avoid disturbing a resolution in progress for an
                     // outer call expression. Effectively we just want a snapshot of whatever has been
                     // inferred for any outer call expression so far.
-                    const instantiatedType = instantiateType(contextualType, cloneTypeMapper(getContextualMapper(node)));
+                    const instantiatedType = instantiateType(contextualType, cloneTypeMapper(getContextualMapper(node), InferenceFlags.NoDefault));

```

Comment:
The `cloneTypeMapper` call used to create the `returnType` mapper inside call resolution doesn't pass `InferenceFlags.NoDefault` with this change. Intentional?

Reply:
Yes, it's intentional. We only want the `InferenceFlags.NoDefault` behavior here (because we're transposing inferences from the outer call mapper and don't want an absence of inferences in the outer mapper to turn into what appears to be a concrete inference of the default).